### PR TITLE
V1.31.0 rebase master

### PR DESCRIPTION
**Pull request by theishshah**
Sync with upstream v1.31.0
 
Done via `./UPSTREAM-MERGE.sh v1.31.0`

### CODE REVIEW

**Review by rashmigottipati — APPROVED** [review]
/lgtm

File: testdata/go/v3/memcached-operator/Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ endif
 
 # Set the Operator SDK version to use. By default, what is installed on the system is used.
 # This is useful for CI or a project to utilize a specific version of the operator-sdk toolkit.
-OPERATOR_SDK_VERSION ?= v1.29.0
+OPERATOR_SDK_VERSION ?= v1.30.0
```
Comment (oceanc80) [inline testdata/go/v3/memcached-operator/Makefile:51]: Should this be 1.31.0?

Comment (rashmigottipati) [reply testdata/go/v3/memcached-operator/Makefile:51]: This version is taken from the binary that is available locally so I think it would be `1.30.0`.

**Comment by KeenonLee** [comment]
operator-sdk version: "v1.30.0-ocp", commit: "1315dfc4d80d8215589e68ef5a49ac7dc9b3a419", kubernetes version: "v1.26.0", go version: "go1.20.4", GOOS: "linux", GOARCH: "amd64"

Should this be 1.31.0?

**Review by everettraven — CHANGES_REQUESTED** [review]
@theishshah it looks like we are missing one change: https://github.com/openshift/ocp-release-operator-sdk/blob/9a49219477ed0f6ddc1c9e5e885cbfedffb51b64/patches/03-setversion.patch#L10 should be `v1.31.0-ocp`

**Review by rashmigottipati — APPROVED** [review]

**Review by everettraven — APPROVED** [review]
/lgtm

**Review by KeenonLee — COMMENTED** [review]
/label qe-approved

**Pull request merged**